src/Interpreters/Cache/EvictionCandidates.cpp

```diff
@@ -121,57 +128,77 @@ void EvictionCandidates::evict()
             continue;
         }
 
+        KeyCandidates failed_key_candidates;
+        failed_key_candidates.key_metadata = key_candidates.key_metadata;
+
         while (!key_candidates.candidates.empty())
         {
             auto & candidate = key_candidates.candidates.back();
-            if (!candidate->releasable())
+            try
```

Comment:
Here inside `try` nothing changes, I only added processing of the exception in the `catch` part.